caldate_fmjd.c

```diff
@@ -0,0 +1,48 @@
+#include "caldate.h"
+
+void caldate_frommjd(cd,day,pwday,pyday)
```

Comment:
We agreed on C89 function signatures for things we introduce or touch anyway. Please convert them, and also add them to the header files.

Reply:
@xenotrope, I kinda goaded you into submitting this PR based on your longstanding patch. Are you interested in the possibly tedious work of responding to review by incorporating suggested changes, rebasing and force-pushing your branch, and awaiting further review? Or would you prefer that someone else take it from here? Either way is totally fine. Either way, this is your work and you'll continue to be its git "author" (unless for some reason you'd rather not be, which we can also accommodate).